components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -916,6 +916,9 @@ void AdsServiceImpl::OnDetectUncertainFuture(const uint32_t number_of_start,
 
   sys_info->is_uncertain_future = is_uncertain_future;
 
+  if (!bat_ads_service_.is_bound()) {
```

Comment:
This would stop ads working, should we not use CHECK?

Reply:
this could happen if a task with `AdsServiceImpl::Start()` was queued here: [link](https://github.com/brave/brave-core/blob/6995a7491a69b8995d126b6dcaf2c4f4c704e3ae/components/brave_ads/browser/ads_service_impl.cc#L791) and `AdsServiceImpl::Shutdown()` is called prior the task execution.

Reply:
I have moved `bat_ads_service_->SetSysInfo` call into `AdsServiceImpl::OnEnsureBaseDirectoryExists` function